### PR TITLE
🔐 Auth functionality

### PR DESCRIPTION
**Pull request by danielbayerlein**
* Define the `authKey` as prop on every widget
* `auth.js` contains the reference (`authKey` + `ENV`)
* In your personal `.env` file you can set the password

### CODE REVIEW

File: components/widgets/bitbucket/pull-request-count.js [hunk 1]
```diff
@@ -40,10 +42,15 @@ export default class BitbucketPullRequestCount extends Component {
   }
 
   async fetchInformation () {
-    const { url, project, repository, users } = this.props
+    const { authKey, url, project, repository, users } = this.props
+    let opts = {}
```
Comment (chrishelgert) [inline components/widgets/bitbucket/pull-request-count.js]: wouldn´t be 
```
const ops = authKey ? { headers: basicAuthHeader(authKey) } : {}
````
a little bit nicer?
or create a helper function that return the opts (we have this check in every widget)

File: components/widgets/elasticsearch/hit-count.js [hunk 2]
```diff
@@ -24,10 +25,15 @@ export default class ElasticsearchHitCount extends Component {
   }
 
   async fetchInformation () {
-    const { url, index, query } = this.props
+    const { authKey, index, query, url } = this.props
```
Comment (chrishelgert) [inline components/widgets/elasticsearch/hit-count.js]: see comment in BitbucketPullRequestCount

File: components/widgets/jenkins/index.js [hunk 3]
```diff
@@ -60,12 +62,17 @@ export default class Jenkins extends Component {
   }
 
   async fetchInformation () {
-    const { jobs, url } = this.props
+    const { authKey, jobs, url } = this.props
+    let opts = {}
```
Comment (chrishelgert) [inline components/widgets/jenkins/index.js]: see comment in BitbucketPullRequestCount

File: components/widgets/jira/issue-count.js [hunk 4]
```diff
@@ -37,10 +39,15 @@ export default class JiraIssueCount extends Component {
   }
 
   async fetchInformation () {
-    const { url, query } = this.props
+    const { authKey, url, query } = this.props
+    let opts = {}
```
Comment (chrishelgert) [inline components/widgets/jira/issue-count.js]: see comment in BitbucketPullRequestCount

File: components/widgets/sonarqube/index.js [hunk 5]
```diff
@@ -71,7 +73,12 @@ export default class SonarQube extends Component {
   }
 
   async fetchInformation () {
-    const { url, componentKey } = this.props
+    const { authKey, url, componentKey } = this.props
+    let opts = {}
```
Comment (chrishelgert) [inline components/widgets/sonarqube/index.js]: see comment in BitbucketPullRequestCount

File: next.config.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const Dotenv = require('dotenv-webpack')
+
+module.exports = {
+  webpack: (config, { dev }) => {
```
Comment (chrishelgert) [inline next.config.js]: you read the variable "dev", but don´t use it anywhere - is this neccessary?

File: components/widgets/bitbucket/pull-request-count.js (hunk 1 above)
Comment (danielbayerlein) [inline components/widgets/bitbucket/pull-request-count.js]: ✅ ternary operator

File: components/widgets/elasticsearch/hit-count.js (hunk 2 above)
Comment (danielbayerlein) [inline components/widgets/elasticsearch/hit-count.js]: ✅ ternary operator

File: components/widgets/jenkins/index.js (hunk 3 above)
Comment (danielbayerlein) [inline components/widgets/jenkins/index.js]: ✅ ternary operator

File: components/widgets/jira/issue-count.js (hunk 4 above)
Comment (danielbayerlein) [inline components/widgets/jira/issue-count.js]: ✅ ternary operator

File: components/widgets/sonarqube/index.js (hunk 5 above)
Comment (danielbayerlein) [inline components/widgets/sonarqube/index.js]: ✅ ternary operator

File: next.config.js (hunk 6 above)
Comment (danielbayerlein) [inline next.config.js]: Copy & Paste from Next.js example ... 

✅ Remove unused param

**Comment by danielbayerlein** [comment]
@chrishelgert Ready for review

**Pull request merged**